lbrynet/core/LBRYWallet.py

```diff
@@ -341,28 +390,15 @@ def _get_stream_info_from_value(self, result, name):
             log.warning("Got an error looking up a name: %s", json.dumps(result))
             return Failure(UnknownNameError(name))
 
-    def claim_name(self, name, bid, sources, metadata, fee=None):
-        value = {'sources': {}}
-        for k in SOURCE_TYPES:
-            if k in sources:
-                value['sources'][k] = sources[k]
-        if value['sources'] == {}:
-            return defer.fail("No source given")
-        for k in BASE_METADATA_FIELDS:
-            if k not in metadata:
-                return defer.fail("Missing required field '%s'" % k)
-            value[k] = metadata[k]
-        for k in metadata:
-            if k not in BASE_METADATA_FIELDS:
-                value[k] = metadata[k]
-        if fee is not None:
-            if "LBC" in fee:
-                value['fee'] = {'LBC': {'amount': fee['LBC']['amount'], 'address': fee['LBC']['address']}}
-
-        d = self._send_name_claim(name, json.dumps(value), bid)
+    def claim_name(self, name, bid, m):
+
+        metadata = Metadata(m)
+
+        d = self._send_name_claim(name, json.dumps(metadata), bid)
 
         def _save_metadata(txid):
-            d = self._save_name_metadata(name, value['sources']['lbry_sd_hash'], txid)
+            log.info("Saving metadata for claim %s" % txid)
```

Comment:
pylint can warn about this, I'll turn it on. log will do the interpolation for you.